clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -359,6 +359,7 @@ private boolean maybeSendTransactionalRequest(long now) {
                     client.send(clientRequest, now);
                     return true;
                 }
+                break; // break the loop if we failed to find a specific node
```

Comment:
I am not sure we always want to do break if we cannot find a node to send the request to. Other than the bootstrapping case, we typically just want to refresh metadata and try again. I think I'd suggest we leave this logic as is, but add a condition to the loop to check for producer shutdown.